src/prefect/engine.py

```diff
@@ -31,16 +28,11 @@
     try:
         from prefect.flow_engine import (
             load_flow_and_flow_run,
-            run_flow_async,
-            run_flow_sync,
+            run_flow,
         )
 
         flow_run, flow = load_flow_and_flow_run(flow_run_id=flow_run_id)
-        # run the flow
-        if flow.isasync:
-            run_coro_as_sync(run_flow_async(flow, flow_run=flow_run))
-        else:
-            run_flow_sync(flow, flow_run=flow_run)
+        run_flow(flow, flow_run=flow_run)
```

Comment:
If the flow is async then this may return a coroutine. We probably need to check if it returns a coroutine and run it synchronously if a coroutine is returned.

Reply:
:+1: @cicdw suggested that I could just let `run_flow` handle it, I think you're right though because `run_flow` isn't doing the coro_as_sync bit.

Reply:
Yup exactly!